stellar-contract-env-common/src/unimplemented_env.rs

```diff
@@ -0,0 +1,86 @@
+use super::{call_macro_with_all_host_functions, Env, EnvBase, RawObj, RawVal};
+use core::any;
+use paste::paste;
+
+#[derive(Clone, Default)]
+pub struct UnimplementedEnv;
+
+impl EnvBase for UnimplementedEnv {
+    fn as_mut_any(&mut self) -> &mut dyn any::Any {
+        self
+    }
+
+    fn check_same_env(&self, _other: &Self) {}
+}
+
+// This is a helper macro used only by generate_env_unimplemented below. It
+// consumes a token-tree of the form:
+//
+//  {fn $fn_id:ident $args:tt -> $ret:ty}
+//
+// and produces the the corresponding method declaration to be used in the Env
+// trait.
+macro_rules! host_function_helper {
+    {fn $fn_id:ident($($arg:ident:$type:ty),*) -> $ret:ty}
+    =>
+    {
+        fn $fn_id(&self, $(paste!{[<_ $arg>]}:$type),*) -> $ret {
```

Comment:
You can just use `_:$type` here and avoid the paste! dep

(if you really want to keep paste, I think it can move to a build-dep not a dep)

Reply:
Ah nice. Good idea. I don't want paste. I was only using it because concat_ident macro is unstable and I couldn't use it.